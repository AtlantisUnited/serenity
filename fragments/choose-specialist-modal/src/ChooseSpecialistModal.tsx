import React                        from 'react'

import { Avatar }                   from '@ui/avatar'
import { Button }                   from '@ui/button'
import { Box, Column, Layout, Row } from '@ui/layout'
import { Modal }                    from '@ui/modal'
import { Text }                     from '@ui/text'

import messages                     from './messages'

export const ChooseSpecialistModal = ({ intl, visible, onClose, onSave, data = {} }: any) => (
  <Modal visible={visible} onClose={onClose} opacity='small'>
    <Box
      maxWidth={['none', 512]}
      width='100%'
      backgroundColor='white'
      borderRadius='extra'
      p='20px 32px 32px'
      m='auto'
      mx={20}
      flexDirection='column'
    >
      <Layout>
        <Text fontSize='massive' lineHeight='medium' fontWeight='semiBold' color='black'>
          {intl.formatMessage(messages.selection)}
        </Text>
      </Layout>
      <Layout flexBasis={10} />
      <Layout>
        <Text fontSize='regular' lineHeight='large' color='black'>
          {intl.formatMessage(messages.selectionDescription)}
        </Text>
      </Layout>
      <Layout flexBasis={30} />
      <Box
        width='100%'
        borderRadius='extra'
        boxShadow='black'
        border='black'
        p={16}
        alignItems='center'
      >
        <Layout>
          <Avatar src={data.photo && data.photo.url} width={56} height={56} color='alto' />
        </Layout>
        <Layout flexBasis={16} />
        <Layout>
          <Column>
            <Layout>
              <Text fontWeight='medium' fontSize='regular' lineHeight='medium' color='black'>
                {data.personalInformation && data.personalInformation.firstName}{' '}
                {data.personalInformation && data.personalInformation.lastName}
              </Text>
            </Layout>
          </Column>
        </Layout>
      </Box>
      <Layout flexBasis={24} />
      <Layout>
        <Row>
          <Layout flexGrow={1}>
            <Button onClick={onClose} size='large' fill>
              {intl.formatMessage(messages.cancel)}
            </Button>
          </Layout>
          <Layout flexBasis={16} />
          <Layout flexGrow={3}>
            <Button onClick={onSave} size='large' color='chicago' fill>
              {intl.formatMessage(messages.assign)}
            </Button>
          </Layout>
        </Row>
      </Layout>
    </Box>
  </Modal>
)
