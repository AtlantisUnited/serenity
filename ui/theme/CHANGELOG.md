# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 0.0.5 (2021-05-10)

**Note:** Version bump only for package @ui/theme

## 0.0.4 (2020-09-28)

### Bug Fixes

- **accounts:** add emotion-server, delete font ([08acf5a](https://github.com/Atlantis-Lab/serenity/commit/08acf5a7169352a606925fdfbbaa18f9917a8106))

## 0.0.3 (2020-06-30)

**Note:** Version bump only for package @ui/theme

## 0.0.2 (2020-06-10)

**Note:** Version bump only for package @ui/theme
