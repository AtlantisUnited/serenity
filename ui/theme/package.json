{
  "name": "@ui/theme",
  "version": "0.0.5",
  "license": "MIT",
  "main": "src/index.ts",
  "dependencies": {
    "@emotion/css": "^11.1.3",
    "react": "^17.0.2",
    "styled-system": "^5.1.0"
  },
  "devDependencies": {
    "@emotion/react": "^11.7.1",
    "@emotion/styled": "^11.6.0",
    "@types/react": "^17.0.6",
    "@types/styled-system": "5.1.4"
  },
  "peerDependencies": {
    "@emotion/react": "*",
    "@emotion/styled": "*"
  }
}
