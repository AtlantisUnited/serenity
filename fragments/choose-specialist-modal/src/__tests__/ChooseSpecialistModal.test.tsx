import React                 from 'react'
import { IntlProvider }      from 'react-intl'

import { ThemeProvider }     from '@ui/theme'
import { render }            from '@testing-library/react'

import ChooseSpecialistModal from '..'

describe('Choose specialist modal fragment', () => {
  describe('snapshots', () => {
    it('should match latest render snapshot', () => {
      const { asFragment } = render(
        <IntlProvider locale='ru' defaultLocale='ru' messages={{}}>
          <ThemeProvider>
            <ChooseSpecialistModal />
          </ThemeProvider>
        </IntlProvider>
      )

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
