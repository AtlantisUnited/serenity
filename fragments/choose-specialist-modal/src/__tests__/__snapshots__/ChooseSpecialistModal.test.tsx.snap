// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choose specialist modal fragment snapshots should match latest render snapshot 1`] = `<DocumentFragment />`;
